src/callbacks_step/analysis_surface_2d.jl

```diff
@@ -0,0 +1,290 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# This file contains callbacks that are performed on the surface like computation of
+# pointwise surface forces.
+
+"""
+    AnalysisSurface{Semidiscretization, Variable}(semi,
+                                                  boundary_symbol_or_boundary_symbols,
+                                                  variable)
+
+This struct is used to compute pointwise surface values of a quantity of interest `variable`
+alongside the boundary/boundaries associated with particular name(s) given in `boundary_symbol`
+or `boundary_symbols`.
+For instance, this can be used to compute the surface pressure coefficient [`SurfacePressureCoefficient`](@ref) or
+surface friction coefficient [`SurfaceFrictionCoefficient`](@ref) of e.g. an airfoil with the boundary
+name `:Airfoil` in 2D.
+
+- `semi::Semidiscretization`: Passed in to retrieve boundary condition information
+- `boundary_symbol_or_boundary_symbols::Symbol|Vector{Symbol}`: Name(s) of the boundary/boundaries
```

Comment:
If we are changing the above to `symbol`, should we change this too?

```suggestion
- `boundary_symbol_or_boundary_symbols::Symbol|Vector{Symbol}`: Name(s) of the boundary/boundaries stored as symbol(s)
```